From ab211383abaaa80a3503ee53f00d801940a80b1f Mon Sep 17 18:57:16 2022
From: Hellosway <hellosway@protonmail.com>
Date: Thu, 17 Mar 2022 18:19:43 -0700
Subject: [PATCH] Enable building UEFI unified kernel image on aarch64.

Makes `mkinitcpio` also look for an aarch64 kernel at `/boot/Image`. Bt default
it is hardcoded to x86 kernels. This also supports aarch64 stub loaders.
---
 mkinitcpio | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mkinitcpio b/mkinitcpio
index 549f381..fbc1e8a 100755
--- a/mkinitcpio
+++ b/mkinitcpio
@@ -270,7 +270,7 @@ build_uefi(){
     msg "Creating UEFI executable: %s" "$out"
 
     if [[ -z "$uefistub" ]]; then
-        for stub in {/usr,}/lib/{systemd/boot/efi,gummiboot}/linux{x64,ia32}.efi.stub; do
+        for stub in {/usr,}/lib/{systemd/boot/efi,gummiboot}/linux{x64,ia32,aa64}.efi.stub; do
             if [[ -f "$stub" ]]; then
                 uefistub="$stub"
                 msg2 "Using UEFI stub: %s" "$uefistub"
@@ -283,7 +283,7 @@ build_uefi(){
     fi
 
     if [[ -z "$kernelimg" ]]; then
-        for img in "/lib/modules/$KERNELVERSION/vmlinuz" "/boot/vmlinuz-$KERNELVERSION" "/boot/vmlinuz-linux"; do
+        for img in "/lib/modules/$KERNELVERSION/vmlinuz" "/boot/vmlinuz-$KERNELVERSION" "/boot/Image" "/boot/vmlinuz-linux"; do
             if [[ -f "$img" ]]; then
                 kernelimg="$img"
                 msg2 "Using kernel image: %s" "$kernelimg"
-- 
2.32.0 (Apple Git-132)

